.github/workflows/deploy-to-gh-pages.yaml

```diff
@@ -0,0 +1,46 @@
+    name: Deploy to GitHub Pages
+
+    on:
+      push:
+        branches:
+          - main
+      pull_request:
+
+    jobs:
+      deploy:
+        runs-on: ubuntu-24.04
+        permissions:
+          contents: write
+        concurrency:
+          group: ${{ github.workflow }}-${{ github.ref }}
+        steps:
+          - uses: actions/checkout@v3
+
+          - name: Setup Node
+            uses: actions/setup-node@v4
+            with:
+              node-version: "20"
+              cache: "npm"
+
+          # - name: Persist npm cache
+          #   uses: actions/cache@v3
+          #   with:
+          #     path: ~/.npm
+          #     key: ${{ runner.os }}-node-${{ hashFiles('**/package.json') }}
+
+          - name: Persist Eleventy .cache
+            uses: actions/cache@v3
+            with:
+              path: ./.cache
+              key: ${{ runner.os }}-eleventy-fetch-cache
+
+          - run: npm install
+          - run: npm run build
+
+          - name: Deploy
+            uses: peaceiris/actions-gh-pages@v3
```

Comment:
Any reason in particular for using a third-party action here?

Github provides several now: [Deploy](https://github.com/marketplace/actions/deploy-github-pages-site), [Configure](https://github.com/marketplace/actions/configure-github-pages), and [Upload Artifact](https://github.com/marketplace/actions/upload-github-pages-artifact).

Reply:
only that this is adapted right from the 11ty docs and thats what they have. I believe that this documentation may have been contributed by them?

Reply:
Alright, let's run with it, but plan to move to the official actions in short order.

Reply:
So it looks like [Configure](https://github.com/actions/configure-pages) only works with Next, Nuxt, Gatsby,  and SvelteKit if you dig into the code [here](https://github.com/actions/configure-pages/blob/main/src/set-pages-config.test.js).